Modules/System/AL Rest Client/src/Authentication/HttpAuthenticationBasic.Codeunit.al

```diff
@@ -0,0 +1,48 @@
+/// <summary>Implementation of the "Http Authentication" interface for a request that requires basic authentication</summary>
+codeunit 2356 "Http Authentication Basic" implements "Http Authentication"
+{
+    var
+        [NonDebuggable]
+        GlobalUsername, GlobalPassword : Text;
+
+    [NonDebuggable]
+    /// <summary>Initializes the authentication object with the given username and password</summary>
+    /// <param name="Username">The username to use for authentication</param>
+    /// <param name="Password">The password to use for authentication</param>
+    procedure Initialize(Username: Text; Password: Text)
```

Comment:
Just a small note: if we don't merge this before v23, we should make use of the new SecretText data type. Or change it afterwards

Reply:
That is an excellent point. We could release without it initially and uptake later though.

Reply:
I've updated the platform version to 23 and runtime version to 12.0 to support SecretText.
Changed Password to SecretText and also an overload function for Http Headers of type SecretText.
Because the password needs to be converted to Base64 for basic authentication, it could potentially leak from the Base64 codeunit. I've removed the dependency on the Base64 module and created a small function to convert SecretText to Base64 while keeping it a SecretText.